### PR TITLE
[swsssdk] update DBInterface redis_client index from db_id to db_name

### PR DESCRIPTION
**Pull request by dzhangalibaba**
* DBInterface stores redis_client with index db_name

Signed-off-by: Dong Zhang d.zhang@alibaba-inc.com

### CODE REVIEW

**Comment by dzhangalibaba** [comment]
@qiluo-msft

File: src/swsssdk/interface.py [hunk 1]
```diff
@@ -159,206 +159,210 @@ def __init__(self, **kwargs):
...
         """
         Connect to database id.
         """
         if db_id is None:
             raise ValueError("No database ID configured for '{}'".format(db_id))
 
+        if db_name is None:
+            raise ValueError("No database Name configured for '{}'".format(db_name))
+
         client = redis.StrictRedis(db=db_id, **self.redis_kwargs)
```
Comment (qiluo-msft) [inline src/swsssdk/interface.py:184]: >redis.StrictRedis [](start = 17, length = 17)

If `self.redis_clients` already has this StrictRedis, we don't need to create a new one.

**Comment by qiluo-msft** [comment]
>             def keys(self, db_id):

db_name

---
Refers to: src/swsssdk/interface.py:21 in 00f0543. [](commit_id = 00f05432785dae4e8bdf6a0b43bfc2787699cb47, deletion_comment = False)

**Review by qiluo-msft — CHANGES_REQUESTED** [review]
As comments

**Comment by dzhangalibaba** [comment]
> > ```
> >         def keys(self, db_id):
> > ```
> 
> db_name
> 
> Refers to: src/swsssdk/interface.py:21 in [00f0543](https://github.com/Azure/sonic-py-swsssdk/commit/00f05432785dae4e8bdf6a0b43bfc2787699cb47). [](commit_id = [00f0543](https://github.com/Azure/sonic-py-swsssdk/commit/00f05432785dae4e8bdf6a0b43bfc2787699cb47), deletion_comment = False)

UPDATED

File: src/swsssdk/interface.py (hunk 1 above)
Comment (dzhangalibaba) [reply src/swsssdk/interface.py]: it is a day 1 behavior, we might consider update all other previous release as well. Anyway we can fix it in dev-master for now. code updated.

**Review by qiluo-msft — APPROVED** [review]
Thanks!

File: src/swsssdk/interface.py [hunk 2]
```diff
@@ -159,206 +159,211 @@ def __init__(self, **kwargs):
...
+    def _connection_error_handler(self, db_name):
         """
         In the event Redis is unavailable, close existing connections, and try again.
         """
         logger.warning('Could not connect to Redis--waiting before trying again.')
-        self.close(db_id)
+        self.close(db_name)
         time.sleep(self.CONNECT_RETRY_WAIT_TIME)
-        self.connect(db_id, True)
+        self.connect(db_name, True)
```
Comment (qiluo-msft) [inline src/swsssdk/interface.py:369]: >self.connect [](start = 8, length = 12)

Number of parameters does not match.

**Comment by imzyxwvu** [comment]
Hello! I saw that you wrapped the statements that turn on `notify-keyspace-events` in `_onetime_connect` from `interface.py` with a `db_name not in self.redis_clients.keys()` codition. Is there any reason behind this change?

**Pull request merged**